Mage.Sets/src/mage/cards/a/AlelaCunningConqueror.java

```diff
@@ -0,0 +1,130 @@
+package mage.cards.a;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbility;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.FirstSpellOpponentsTurnTriggeredAbility;
+import mage.abilities.effects.common.CreateTokenEffect;
+import mage.abilities.effects.common.combat.GoadTargetEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.events.DamagedBatchEvent;
+import mage.game.events.DamagedEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.FaerieRogueToken;
+import mage.players.Player;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author Susucr
+ */
+public final class AlelaCunningConqueror extends CardImpl {
+
+    public AlelaCunningConqueror(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{B}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.FAERIE);
+        this.subtype.add(SubType.WARLOCK);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Whenever you cast your first spell during each opponent's turn, create a 1/1 black Faerie Rogue creature token with flying.
+        this.addAbility(new FirstSpellOpponentsTurnTriggeredAbility(
+                new CreateTokenEffect(new FaerieRogueToken()), false
+        ));
+
+        // Whenever one or more Faeries you control deal combat damage to a player, goad target creature that player controls.
+        TriggeredAbility trigger = new AlelaCunningConquerorTriggeredAbility();
+        trigger.addTarget(new TargetCreaturePermanent()); // will be adjusted on trigger.
```

Comment:
what's the point of adding a target here if it'll be cleared and added in checkTrigger?

Reply:
For the rules text generation mostly. I can try if everything is well by setting text instead.

Reply:
I'm pretty sure it's not relevant if you're overriding getRule